### PR TITLE
Vritant/refactor pool updates

### PR DESCRIPTION
**Pull request by vritant**
Tackled a couple of TODOs from my previous PR
* skip refresh pools when you can
* refactor update pools - dont depend on update subscriptions any more.

### CODE REVIEW

**Comment by vritant** [comment]
@candlepin-pull-request-bot retest this please

**Comment by Ceiu** [comment]
@candlepin-pull-request-bot retest this please

**Comment by Ceiu** [comment]
Looks like the issue was a hosed CRL on my end from testing. I imagine the tests will pass again.
Going to give this another once-over before ACKing. There's a lot of changes here that could have some particularly nasty effects.

File: server/src/main/java/org/candlepin/model/dto/Subscription.java [hunk 1]
```diff
@@ -86,8 +86,14 @@ public Subscription(Owner ownerIn, Product productIn, Set<Product> providedProdu
     }
 
     public String toString() {
-        return "Subscription [id = " + getId() + ", product = " + getProduct().getId() +
-            ", owner = " + getOwner().getKey() + "]";
+        String subscription = "Subscription [id = " + getId();
+        if (product != null) {
+            subscription += getProduct().getId();
```
Comment (cnsnyder) [inline server/src/main/java/org/candlepin/model/dto/Subscription.java]: Should this line be: `subscription += ", product = " + getProduct().getId();`

**Comment by cnsnyder** [comment]
I've looked it over twice now. It seems to make sense to me (aside from the one small change requested). I think this should be good to merge with that change.

**Comment by vritant** [comment]
@cnsnyder done!

**Comment by cnsnyder** [comment]
Looks good, ACK.

**Pull request merged**